enrichissement.py

```diff
@@ -49,10 +49,126 @@ def main():
+                     "acheteurNom" : "nomAcheteur",
+                     "natureObjet": "natureObjetMarche",
+                     "categorieEntreprise" : "categorieEtablissement"
+                     })
+    
+    df = df.drop(columns = ['latitudeCommuneEtablissement', 'longitudeCommuneEtablissement', 'latitudeCommuneAcheteur', 'longitudeCommuneAcheteur'])
+
+    df = df[['id', 'source', 'type', 'natureObjetMarche', 'objetMarche', 'codeCPV', 'referenceCPV', 'dureeMois',
+        'dateNotification', 'anneeNotification', 'moisNotification', 'dureeMoisEstime',
+        'dureeMoisCalculee', 'datePublicationDonnees','montantOriginal', 'montantEstime', 'montantCalcule', 'nbTitulairesSurCeMarche',
+        'formePrix', 'lieuExecutionCode', 'lieuExecutionTypeCode',
+        'lieuExecutionNom', 'nature', 'procedure',
+        'idAcheteur', 'verifSirenAcheteur', 'nomAcheteur', 'codeRegionAcheteur', 'regionAcheteur', 'codePostalAcheteur', 'libelleCommuneAcheteur', 'codeCommuneAcheteur',
+        'superficieCommuneAcheteur', 'populationCommuneAcheteur', 'geolocCommuneAcheteur',
+        'typeIdentifiantEtablissement',
+        'siretEtablissement', 'sirenEtablissement', 'nicEtablissement', 'verifSiretEtablissement', "categorieEtablissement", 'denominationSocialeEtablissement',
+        'adresseEtablissement', 'communeEtablissement', 'codeCommuneEtablissement', 'codePostalEtablissement',
+        'codeTypeEtablissement', 
+        'superficieCommuneEtablissement', 'populationCommuneEtablissement',
+        'distanceAcheteurEtablissement', 
+        'geolocCommuneEtablissement']]
+    return df
+
+
+def enrichissement_type_entreprise(df):
+    #Recuperation de la base 
+    path = os.path.join(path_to_data, conf["base_type_entreprise"])
+    to_add = pd.read_csv(path, usecols = ["siren", "categorieEntreprise", "nicSiegeUniteLegale"])
+    #On doit creer Siret
+    to_add["NIC"] = to_add.apply(lambda x: ("00000" + str(x["nicSiegeUniteLegale"]))[-5:], axis=1)
+    to_add["siretEtablissement"] = to_add.apply(lambda x: str(x["siren"]) + str(x["NIC"]), axis=1)    
+    #Jointure sur le Siret entre df et to_add
+    #On vérifie que le code siret n'est qu en un exemplaire dans la base
+    try :
+        df = suppression_Siret_doublon(df)
+        df = df.merge(to_add[['categorieEntreprise','siretEtablissement']], how = 'left', on = 'siretEtablissement')
+        return df
+    except:
+        df = df.merge(to_add[['categorieEntreprise','siretEtablissement']], how = 'left', on = 'siretEtablissement')
+        return df        
+
+
+def suppression_Siret_doublon(df):
+    Siret = df["siretEtablissement"]
+    Siret = Siret.iloc[:,0]
+    df = df.drop(columns = ["siretEtablissement"])
+    df["siretEtablissement"] = Siret
+    return df
+
+
+##### Algorithme de Luhn
+
+def luhn(nbentier):
+    """Application de la formule de Luhn à un nombre"""
+    try :
+        chiffres = pd.DataFrame(map(int, list(str(nbentier))), columns=['Number_list'])
```

Comment:
une liste serait plus adaptee qu'un dataframe

Reply:
Je suggere ce code, un peu plus court :
def is_luhn_valid(x):
    luhn_corr = [0,2,4,6,8,1,3,5,7,9]
    l = [int(i) for i in list(str(x))]
    l2 = [luhn_corr[i] if (index+1) % 2 == 0 else i for index, i in enumerate(l[::-1])]
    return sum(l2) % 10 == 0